qiskit_machine_learning/gradients/param_shift/param_shift_sampler_gradient.py

```diff
@@ -89,20 +95,40 @@ def _run_unique(
             job_circuits.extend([circuit] * n)
             job_param_values.extend(param_shift_parameter_values)
             all_n.append(n)
+        cirs_params = [(job_circuits[i],job_param_values[i]) for i in range(n)]
 
         # Run the single job with all circuits.
-        job = self._sampler.run(job_circuits, job_param_values, **options)
+        if isinstance(self._sampler, BaseSamplerV1):
+            job = self._sampler.run(job_circuits, job_param_values, **options)
+        elif isinstance(self._sampler, BaseSamplerV2):
+            job = self._sampler.run(cirs_params)
+        else:
+            raise AlgorithmError("Wrong Sampler Type.") 
```

Comment:
```suggestion
            raise AlgorithmError(f"The accepted estimators are BaseSamplerV1 (deprecated) and BaseSamplerV2; got {type(self._sampler)} instead.") 
```
As for the Estimator case.